validators.py

```diff
@@ -80,3 +80,11 @@ def validate(value: Iterable) -> None:
                            f'unexpected values: {value_as_set.difference(expected_set)}'))
 
     return validate
+
+
+def no_whitespace_check():
+    def validate(value):
+        if value.startswith(" "):
```

Comment:
I'm not sure if `startsWith` qualifies as "contains whitespace". I think we should compare the original with the `.strip()` result if you want to know whether there was whitespace at the beginning or the end of the value.

You can use `value.replace(" ", "") == ''` to check to see if a value is _entirely_ whitespace.